x/evm/types/journal.go

```diff
@@ -204,11 +204,16 @@ func (ch refundChange) dirtied() *ethcmn.Address {
 }
 
 func (ch addLogChange) revert(s *CommitStateDB) {
-	logs := s.logs[ch.txhash]
+	logs, err := s.GetLogs(ch.txhash)
+	if err != nil {
+		// panic on unmarshal error
+		panic(err)
```

Comment:
hm... `panic` seems intense

Reply:
This is an internal error that is only present is there's an unmarshal error. We want to panic here bc otherwise we could persist invalid state and have to manually stop the chain (which requires coordination of +1/3 validatiors) and then roll-back to the height before the first error instance of this error occurred.

Having a panic prevents this overhead and allows us to detect these errors during development 🙂